### DISCUSSION TITLE
Bugs in Raffle.s.sol and Raffle.t.sol

### DISCUSSION THREAD

**Discussion by igboBoydev** [Cyfrin/foundry-full-course-cu#1599] [Q&A]
Dear teammates, 

I have an issue compiling my raffle.s.sol and raffle.t.sol code bases because of some errors showing up on my IDE. I have carefully looked through the course to see if there were code snippets I missed but I didn't find any so I thought I should reach out and seek help.

Below are pictures of the files in which the errors are coming from.

<img width="1440" alt="Screenshot 2024-04-13 at 9 09 55 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/78387050/916a8477-1af2-421e-be7b-dd71e040aa0e">
<img width="1440" alt="Screenshot 2024-04-13 at 9 10 00 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/78387050/80d1cccc-4e6b-43fb-8c15-23b613f08959">

**Comment by EngrPips** [comment]
Hello @igboBoydev , The terminal clearly stated what the error is and that should be really helpful to you in debugging it

First Error
` Type contract Raffle is not implicitly convertible to expected type tuple(contract Raffle, contract HelperConfig)` 
This simply mean that in your `DeployRaffle` script you didn't return a tuple but you are trying to get out a type of tuple from it. You should show your `DeployRaffle` script if you don't understand what i pointed out.

Second Error
`Different number of components on the left hand side (7) than on the right hand side (6)`
This simply mean you taking out more variable out from the `helperConfig.activeNetworksConfig()` than you have originally designed it to hold, if you don't understand what i point out please show your `HelperConfig` script

Looking forward to hear back from you.

**Reply by igboBoydev** [reply]
Alright I'll share the DeployRaffle Script file as well as the HelperConfig>

Thanks though for the swift response.

**Comment by igboBoydev** [comment]
this is the content of my DeployRaffle Script
```
`// Layout of Contract:
// version
// imports
// errors
// interfaces, libraries, contracts
// Type declarations
// State variables
// Events
// Modifiers
// Functions

// Layout of Functions:
// constructor
// receive function (if exists)
// fallback function (if exists)
// external
// public
// internal
// private
// view & pure functions


// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {VRFCoordinatorV2Interface} from "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
import {VRFConsumerBaseV2} from "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";

/**
 * @title A sample Raffle contract
 * @author Abel Kalu
 * @notice This contract is for creating a sample raffle
 * @dev Implements Chainlink VRFv2
 */
contract Raffle is VRFConsumerBaseV2 {
    error Raffle_NotEnoughEthSent();
    error Raffle_StateNotOpen();
    error Raffle__TransferFailed();
    error Raffle_UpkeepNotNeeded(uint256 currentBalance,  uint256 numPlayers, uint256 raffleState);

    enum RaffleState {
        OPEN, 
        CALCULATING
    }

    uint16 private constant REQUEST_CONFIRMATIONS = 3;
    uint32 private constant NUM_WORDS = 1;

    uint256 private immutable i_entranceFee;
    uint32 private immutable i_callbackGasLimit;
    uint256 private immutable i_interval;
    uint64 private immutable i_subscriptionId;
    VRFCoordinatorV2Interface private immutable i_vrfCoordinator;
    bytes32 private immutable i_gasLane;
    RaffleState private s_raffleState;

    uint256 private s_lastTimeStamp;
    address private s_recentWinner;
    address payable[] private s_players;

    event EnteredRaffle(address indexed player);
    event WinnerPicked(address indexed winner);

    constructor(
        uint256 entranceFee,
        uint256 interval,
        address vrfCoordinator,
        bytes32 gasLane,
        uint64 subscriptionId,
        uint32 callbackGasLimit
    ) VRFConsumerBaseV2(vrfCoordinator) {
        i_entranceFee = entranceFee;
        i_interval = interval;
        i_gasLane = gasLane;
        s_lastTimeStamp = block.timestamp;
        i_vrfCoordinator = VRFCoordinatorV2Interface(vrfCoordinator);
        i_callbackGasLimit = callbackGasLimit;
        s_raffleState = RaffleState.OPEN;
        i_subscriptionId = subscriptionId;
    }

    function enterRaffle() external payable {
        if (msg.value >= i_entranceFee) {
            revert Raffle_NotEnoughEthSent();
        }
        if(s_raffleState != RaffleState.OPEN){
            revert Raffle_StateNotOpen();
        }
        s_players.push(payable(msg.sender));
        // require(msg.value >= i_entranceFee, "Not enough eth sent");
        emit EnteredRaffle(msg.sender);
    }


    /**
     * @dev this is the function that the Chainlink Automation nodes call to see if it's time to perform, upkeep
     * The following should be true for this to return true
     * 1. The time interval has passed between raffle runs
     * 2. The raffle is in the OPEN state
     * 3. The contract has ETH (aka, players)
     * 4. (Implicit) The subscription is funded with LINK
     */
    
    function checkUpKeep(bytes memory)public view returns (bool upKeepNeeded, bytes memory) {
        bool timeHasPassed = (block.timestamp - s_lastTimeStamp) >= i_interval;
        bool isOpen = RaffleState.OPEN == s_raffleState;
        bool hasBalance = address(this).balance > 0;
        bool hasPlayers = s_players.length > 0;
        upKeepNeeded = (timeHasPassed && isOpen && hasBalance && hasPlayers);
        return (upKeepNeeded, "0x0");
    }

    // 1. Get a random number
    // 2. Use the random number to pick a winner
    // 3. be automatically called
    function performUpkeep(bytes calldata /* performData */) external {
        // check to see if enough time has passed
        // if ((block.timestamp - s_lastTimeStamp) < i_interval) {
        //     revert();
        // }
        (bool upkeepNeeded, ) = checkUpKeep("");
        if(!upkeepNeeded){
            revert Raffle_UpkeepNotNeeded(
                address(this).balance,
                s_players.length,
                uint256(s_raffleState)
            );
        }

        s_raffleState = RaffleState.CALCULATING; 

        i_vrfCoordinator.requestRandomWords(
            i_gasLane,
            i_subscriptionId,
            REQUEST_CONFIRMATIONS,
            i_callbackGasLimit,
            NUM_WORDS
        );
        // 1. Request the RNG
        // Get a random number
    }

    function getEntranceFee() external view returns (uint256) {
        return i_entranceFee;
    }

    function getRaffleState() external view returns (RaffleState) {
        return s_raffleState;
    }

    function fulfillRandomWords(
        uint256  /*requestId */,
        uint256[] memory randomWords
    ) internal virtual override {
        uint256 indexOfWinner = randomWords[0] % s_players.length;
        address payable winner = s_players[indexOfWinner];
        s_recentWinner = winner;
        s_raffleState = RaffleState.OPEN;
        s_players = new address payable[](0);
        s_lastTimeStamp = block.timestamp;
        (bool success, ) = winner.call{value: address(this).balance}("");
        if(!success){
            revert Raffle__TransferFailed();
        }
        emit WinnerPicked(winner);
    }
}
`
```

**Reply by EngrPips** [reply]
This is the `Raffle` contract itself and not the `DeployRaffle` script, please share the `DeployRaffle` script

**Comment by igboBoydev** [comment]
This is the content of my HelperConfig


`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;


import {Script} from "forge-std/Script.sol";
import {VRFCoordinatorV2Mock} from "@chainlink/contracts/src/v0.8/mocks/VRFCoordinatorV2Mock.sol";


contract HelperConfig is Script {
    struct NetworksConfig {
        uint256 entranceFee;
        uint256 interval;
        address vrfCoordinator;
        bytes32 gasLane;
        uint64 subscriptionId;
        uint32 callbackGasLimit;
    }

    NetworksConfig public activeNetworksConfig;

    constructor(){
        if(block.chainid == 11155111){
            activeNetworksConfig = getSapoliaEthConfig();
         }else{
            activeNetworksConfig =  getOrCreateAnvilEthConfig();
         }
    }

    function getSapoliaEthConfig() public pure returns (NetworksConfig memory) {
        return NetworksConfig({
            entranceFee: 0.01 ether,
            interval: 30,
            vrfCoordinator: 0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625,
            gasLane: "0x0",
            subscriptionId: 0,
            callbackGasLimit: 500000
        });
    }

    function getOrCreateAnvilEthConfig() public returns (NetworksConfig memory){
        if(activeNetworksConfig.vrfCoordinator != address(0)){
            return activeNetworksConfig;
        }

        uint96 baseFee = 0.25 ether;
        uint96 gasPriceLink = 1e9;

        vm.startBroadcast();
        VRFCoordinatorV2Mock vrfCoordinatorMock = new VRFCoordinatorV2Mock(
            baseFee,
            gasPriceLink
        );
        vm.stopBroadcast();

        return NetworksConfig({
            entranceFee: 0.01 ether,
            interval: 30,
            vrfCoordinator: address(vrfCoordinatorMock),
            gasLane: "0x0",
            subscriptionId: 0,
            callbackGasLimit: 500000
        });
    }
}
`

**Reply by EngrPips** [reply]
```javascript
`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;

import {Script} from "forge-std/Script.sol";
import {VRFCoordinatorV2Mock} from "@chainlink/contracts/src/v0.8/mocks/VRFCoordinatorV2Mock.sol";

contract HelperConfig is Script {
struct NetworksConfig {
uint256 entranceFee;
uint256 interval;
address vrfCoordinator;
bytes32 gasLane;
uint64 subscriptionId;
uint32 callbackGasLimit;
}

NetworksConfig public activeNetworksConfig;

constructor(){
    if(block.chainid == 11155111){
        activeNetworksConfig = getSapoliaEthConfig();
     }else{
        activeNetworksConfig =  getOrCreateAnvilEthConfig();
     }
}

function getSapoliaEthConfig() public pure returns (NetworksConfig memory) {
    return NetworksConfig({
        entranceFee: 0.01 ether,
        interval: 30,
        vrfCoordinator: 0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625,
        gasLane: "0x0",
        subscriptionId: 0,
        callbackGasLimit: 500000
    });
}

function getOrCreateAnvilEthConfig() public returns (NetworksConfig memory){
    if(activeNetworksConfig.vrfCoordinator != address(0)){
        return activeNetworksConfig;
    }

    uint96 baseFee = 0.25 ether;
    uint96 gasPriceLink = 1e9;

    vm.startBroadcast();
    VRFCoordinatorV2Mock vrfCoordinatorMock = new VRFCoordinatorV2Mock(
        baseFee,
        gasPriceLink
    );
    vm.stopBroadcast();

    return NetworksConfig({
        entranceFee: 0.01 ether,
        interval: 30,
        vrfCoordinator: address(vrfCoordinatorMock),
        gasLane: "0x0",
        subscriptionId: 0,
        callbackGasLimit: 500000
    });
}
```

**Comment by EngrPips** [comment]
> Dear teammates,
> 
> I have an issue compiling my raffle.s.sol and raffle.t.sol code bases because of some errors showing up on my IDE. I have carefully looked through the course to see if there were code snippets I missed but I didn't find any so I thought I should reach out and seek help.
> 
> Below are pictures of the files in which the errors are coming from.
> 
> 

Please remove the `,` after the `uint32 callbackGasLimit` in  the line where you pulled from the `helperConfig.activeNetworksConfig`

**Reply by igboBoydev** [reply]
Hello @EngrPips I really appreciate the help, I am mostly grateful for this.

Thank you.

**Reply by EngrPips** [reply]
You are welcome buddy and you can as well mark the discussion as answered if we have successfully solve your issue

**Reply by igboBoydev** [reply]
Ok I will but my testRaffleInitializesInOpenState test isn't passing yet.

Don't know if I can get some help with that please

**Reply by EngrPips** [reply]
What error is it failing with ?

run the test with the `-vvvv` flag and show the output maybe we can figure out why it is failing

**Comment by igboBoydev** [comment]
Ok doing now

**Reply by igboBoydev** [reply]
`
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {DeployRaffle} from "../../script/Raffle.s.sol";
import {Raffle} from "../../src/Raffle.sol";
import {Test, console} from "forge-std/Test.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";

contract ReffleTest is Test {
    Raffle raffle;
    HelperConfig helperConfig;
    uint256 entranceFee;
    uint256 interval;
    address vrfCoordinator;
    bytes32 gasLane;
    uint64 subscriptionId;
    uint32 callbackGasLimit;

    address public PLAYER = makeAddr("player");
    uint256 public constant STARTING_USER_BALANCE = 10 ether;

    function setUp() external {
        DeployRaffle deployer = new DeployRaffle();
        (raffle) = deployer.run();
         (
        uint256 entranceFee,
        uint256 interval,
        address vrfCoordinator,
        bytes32 gasLane,
        uint64 subscriptionId,
        uint32 callbackGasLimit
        ) = helperConfig.activeNetworksConfig();
    }

    function testRaffleInitializesInOpenState() public view {
        assert(raffle.getRaffleState() == Raffle.RaffleState.OPEN);
    }
}
`

**Reply by igboBoydev** [reply]
raffel.sol


`// Layout of Contract:
// version
// imports
// errors
// interfaces, libraries, contracts
// Type declarations
// State variables
// Events
// Modifiers
// Functions

// Layout of Functions:
// constructor
// receive function (if exists)
// fallback function (if exists)
// external
// public
// internal
// private
// view & pure functions


// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {VRFCoordinatorV2Interface} from "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
import {VRFConsumerBaseV2} from "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";

/**
 * @title A sample Raffle contract
 * @author Abel Kalu
 * @notice This contract is for creating a sample raffle
 * @dev Implements Chainlink VRFv2
 */
contract Raffle is VRFConsumerBaseV2 {
    error Raffle_NotEnoughEthSent();
    error Raffle_StateNotOpen();
    error Raffle__TransferFailed();
    error Raffle_UpkeepNotNeeded(uint256 currentBalance,  uint256 numPlayers, uint256 raffleState);

    enum RaffleState {
        OPEN, 
        CALCULATING
    }

    uint16 private constant REQUEST_CONFIRMATIONS = 3;
    uint32 private constant NUM_WORDS = 1;

    uint256 private immutable i_entranceFee;
    uint32 private immutable i_callbackGasLimit;
    uint256 private immutable i_interval;
    uint64 private immutable i_subscriptionId;
    VRFCoordinatorV2Interface private immutable i_vrfCoordinator;
    bytes32 private immutable i_gasLane;
    RaffleState private s_raffleState;

    uint256 private s_lastTimeStamp;
    address private s_recentWinner;
    address payable[] private s_players;

    event EnteredRaffle(address indexed player);
    event WinnerPicked(address indexed winner);

    constructor(
        uint256 entranceFee,
        uint256 interval,
        address vrfCoordinator,
        bytes32 gasLane,
        uint64 subscriptionId,
        uint32 callbackGasLimit
    ) VRFConsumerBaseV2(vrfCoordinator) {
        i_entranceFee = entranceFee;
        i_interval = interval;
        i_gasLane = gasLane;
        s_lastTimeStamp = block.timestamp;
        i_vrfCoordinator = VRFCoordinatorV2Interface(vrfCoordinator);
        i_callbackGasLimit = callbackGasLimit;
        s_raffleState = RaffleState.OPEN;
        i_subscriptionId = subscriptionId;
    }

    function enterRaffle() external payable {
        if (msg.value >= i_entranceFee) {
            revert Raffle_NotEnoughEthSent();
        }
        if(s_raffleState != RaffleState.OPEN){
            revert Raffle_StateNotOpen();
        }
        s_players.push(payable(msg.sender));
        // require(msg.value >= i_entranceFee, "Not enough eth sent");
        emit EnteredRaffle(msg.sender);
    }


    /**
     * @dev this is the function that the Chainlink Automation nodes call to see if it's time to perform, upkeep
     * The following should be true for this to return true
     * 1. The time interval has passed between raffle runs
     * 2. The raffle is in the OPEN state
     * 3. The contract has ETH (aka, players)
     * 4. (Implicit) The subscription is funded with LINK
     */
    
    function checkUpKeep(bytes memory)public view returns (bool upKeepNeeded, bytes memory) {
        bool timeHasPassed = (block.timestamp - s_lastTimeStamp) >= i_interval;
        bool isOpen = RaffleState.OPEN == s_raffleState;
        bool hasBalance = address(this).balance > 0;
        bool hasPlayers = s_players.length > 0;
        upKeepNeeded = (timeHasPassed && isOpen && hasBalance && hasPlayers);
        return (upKeepNeeded, "0x0");
    }

    // 1. Get a random number
    // 2. Use the random number to pick a winner
    // 3. be automatically called
    function performUpkeep(bytes calldata /* performData */) external {
        // check to see if enough time has passed
        // if ((block.timestamp - s_lastTimeStamp) < i_interval) {
        //     revert();
        // }
        (bool upkeepNeeded, ) = checkUpKeep("");
        if(!upkeepNeeded){
            revert Raffle_UpkeepNotNeeded(
                address(this).balance,
                s_players.length,
                uint256(s_raffleState)
            );
        }

        s_raffleState = RaffleState.CALCULATING; 

        i_vrfCoordinator.requestRandomWords(
            i_gasLane,
            i_subscriptionId,
            REQUEST_CONFIRMATIONS,
            i_callbackGasLimit,
            NUM_WORDS
        );
        // 1. Request the RNG
        // Get a random number
    }

    function getEntranceFee() external view returns (uint256) {
        return i_entranceFee;
    }

    function getRaffleState() external view returns (RaffleState) {
        return s_raffleState;
    }

    function fulfillRandomWords(
        uint256  /*requestId */,
        uint256[] memory randomWords
    ) internal virtual override {
        uint256 indexOfWinner = randomWords[0] % s_players.length;
        address payable winner = s_players[indexOfWinner];
        s_recentWinner = winner;
        s_raffleState = RaffleState.OPEN;
        s_players = new address payable[](0);
        s_lastTimeStamp = block.timestamp;
        (bool success, ) = winner.call{value: address(this).balance}("");
        if(!success){
            revert Raffle__TransferFailed();
        }
        emit WinnerPicked(winner);
    }
}
`

**Reply by EngrPips** [reply]
```javascript
// Layout of Contract:
// version
// imports
// errors
// interfaces, libraries, contracts
// Type declarations
// State variables
// Events
// Modifiers
// Functions

// Layout of Functions:
// constructor
// receive function (if exists)
// fallback function (if exists)
// external
// public
// internal
// private
// view & pure functions

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {VRFCoordinatorV2Interface} from "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
import {VRFConsumerBaseV2} from "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";

/**

@title A sample Raffle contract

https://github.com/author Abel Kalu

@notice This contract is for creating a sample raffle

@dev Implements Chainlink VRFv2
*/
contract Raffle is VRFConsumerBaseV2 {
error Raffle_NotEnoughEthSent();
error Raffle_StateNotOpen();
error Raffle__TransferFailed();
error Raffle_UpkeepNotNeeded(uint256 currentBalance, uint256 numPlayers, uint256 raffleState);

enum RaffleState {
OPEN,
CALCULATING
}

uint16 private constant REQUEST_CONFIRMATIONS = 3;
uint32 private constant NUM_WORDS = 1;

uint256 private immutable i_entranceFee;
uint32 private immutable i_callbackGasLimit;
uint256 private immutable i_interval;
uint64 private immutable i_subscriptionId;
VRFCoordinatorV2Interface private immutable i_vrfCoordinator;
bytes32 private immutable i_gasLane;
RaffleState private s_raffleState;

uint256 private s_lastTimeStamp;
address private s_recentWinner;
address payable[] private s_players;

event EnteredRaffle(address indexed player);
event WinnerPicked(address indexed winner);

constructor(
uint256 entranceFee,
uint256 interval,
address vrfCoordinator,
bytes32 gasLane,
uint64 subscriptionId,
uint32 callbackGasLimit
) VRFConsumerBaseV2(vrfCoordinator) {
i_entranceFee = entranceFee;
i_interval = interval;
i_gasLane = gasLane;
s_lastTimeStamp = block.timestamp;
i_vrfCoordinator = VRFCoordinatorV2Interface(vrfCoordinator);
i_callbackGasLimit = callbackGasLimit;
s_raffleState = RaffleState.OPEN;
i_subscriptionId = subscriptionId;
}

function enterRaffle() external payable {
if (msg.value >= i_entranceFee) {
revert Raffle_NotEnoughEthSent();
}
if(s_raffleState != RaffleState.OPEN){
revert Raffle_StateNotOpen();
}
s_players.push(payable(msg.sender));
// require(msg.value >= i_entranceFee, "Not enough eth sent");
emit EnteredRaffle(msg.sender);
}

/**

@dev this is the function that the Chainlink Automation nodes call to see if it's time to perform, upkeep
The following should be true for this to return true
The time interval has passed between raffle runs
The raffle is in the OPEN state
The contract has ETH (aka, players)
(Implicit) The subscription is funded with LINK
*/
function checkUpKeep(bytes memory)public view returns (bool upKeepNeeded, bytes memory) {
bool timeHasPassed = (block.timestamp - s_lastTimeStamp) >= i_interval;
bool isOpen = RaffleState.OPEN == s_raffleState;
bool hasBalance = address(this).balance > 0;
bool hasPlayers = s_players.length > 0;
upKeepNeeded = (timeHasPassed && isOpen && hasBalance && hasPlayers);
return (upKeepNeeded, "0x0");
}

// 1. Get a random number
// 2. Use the random number to pick a winner
// 3. be automatically called
function performUpkeep(bytes calldata /* performData */) external {
// check to see if enough time has passed
// if ((block.timestamp - s_lastTimeStamp) < i_interval) {
// revert();
// }
(bool upkeepNeeded, ) = checkUpKeep("");
if(!upkeepNeeded){
revert Raffle_UpkeepNotNeeded(
address(this).balance,
s_players.length,
uint256(s_raffleState)
);
}

 s_raffleState = RaffleState.CALCULATING; 

 i_vrfCoordinator.requestRandomWords(
     i_gasLane,
     i_subscriptionId,
     REQUEST_CONFIRMATIONS,
     i_callbackGasLimit,
     NUM_WORDS
 );
 // 1. Request the RNG
 // Get a random number
}

function getEntranceFee() external view returns (uint256) {
return i_entranceFee;
}

function getRaffleState() external view returns (RaffleState) {
return s_raffleState;
}

function fulfillRandomWords(
uint256 /*requestId */,
uint256[] memory randomWords
) internal virtual override {
uint256 indexOfWinner = randomWords[0] % s_players.length;
address payable winner = s_players[indexOfWinner];
s_recentWinner = winner;
s_raffleState = RaffleState.OPEN;
s_players = new address payable;
s_lastTimeStamp = block.timestamp;
(bool success, ) = winner.call{value: address(this).balance}("");
if(!success){
revert Raffle__TransferFailed();
}
emit WinnerPicked(winner);
}
}
`
```

**Reply by EngrPips** [reply]
```javascript
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {DeployRaffle} from "../../script/Raffle.s.sol";
import {Raffle} from "../../src/Raffle.sol";
import {Test, console} from "forge-std/Test.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";

contract ReffleTest is Test {
Raffle raffle;
HelperConfig helperConfig;
uint256 entranceFee;
uint256 interval;
address vrfCoordinator;
bytes32 gasLane;
uint64 subscriptionId;
uint32 callbackGasLimit;

address public PLAYER = makeAddr("player");
uint256 public constant STARTING_USER_BALANCE = 10 ether;

function setUp() external {
    DeployRaffle deployer = new DeployRaffle();
    (raffle) = deployer.run();
     (
    uint256 entranceFee,
    uint256 interval,
    address vrfCoordinator,
    bytes32 gasLane,
    uint64 subscriptionId,
    uint32 callbackGasLimit
    ) = helperConfig.activeNetworksConfig();
}

function testRaffleInitializesInOpenState() public view {
    assert(raffle.getRaffleState() == Raffle.RaffleState.OPEN);
}
}
```

**Reply by EngrPips** [reply]
@igboBoydev  please run the test and make sure to use the `-vvvv` flag and then show me the result of the test, i mean the output you get when you run the test
